src/intervals/interval_operations/constants.jl

```diff
@@ -4,40 +4,43 @@
 """
     emptyinterval
 
-`emptyinterval`s are represented as the interval [∞, -∞]; note
-that this interval is an exception to the fact that the lower bound is
-larger than the upper one.
-
-Note that if the type of the returned interval can not be inferred from the
-argument given, the default interval bound type is used.
+Create an empty interval. This interval is an exception to the fact that the
+lower bound is larger than the upper one.
 
 Implement the `empty` function of the IEEE Standard 1788-2015 (Section 10.5.2).
 """
-emptyinterval(::Type{Interval{T}}) where {T<:NumTypes} = unsafe_interval(T, typemax(T), typemin(T))
-emptyinterval(::Type{T}) where {T<:NumTypes} = emptyinterval(Interval{T})
-emptyinterval(::Type{<:Real}) = emptyinterval(default_numtype())
-emptyinterval() = emptyinterval(default_numtype())
+emptyinterval(::Type{BareInterval{T}}) where {T<:AbstractFloat} = _unsafe_bareinterval(T, convert(T, NaN), convert(T, NaN))
+emptyinterval(::Type{BareInterval{T}}) where {T<:Rational} = _unsafe_bareinterval(T, typemax(T), typemin(T))
+# note: `using Base.unsafe_rational(Int, 0, 0)` as an equivalent to `NaN` for `Rational`
+# does not work well since most codes for `Rational` assume that the denominator cannot be zero
+# e.g. `iszero(Base.unsafe_rational(Int, 0, 0)) == true`
```

Comment:
I think this is worth reporting to Julia!

Reply:
Good idea! I will do this over the week-end.